config.yaml

```diff
@@ -20,8 +20,10 @@ options:
     description: OpenID Connect client secret
   public-url:
     type: string
-    default: ''
-    description: Publicly-accessible endpoint for cluster
+    default: 'http://dex-auth.kubeflow.svc.cluster.local:5556'
```

Comment:
Can this be computed a bit more smartly? This assumes that the deployed application is named 'dex-auth' (and that the model name is 'kubeflow')

Could we not use lightkube to inspect the address of the loadbalancer service for istio ingress and set this automatically to the correct `x.x.x.x.nip.io` address? 

While this is a more useful default than nothing, in reality people will still need to adjust it to access externally, no?

Reply:
(I couldn't decide whether to leave this comment here or on the other PR for dex-auth but it sort of applies to both)

Reply:
@jnsgruk I think there's a misunderstanding here, which is caused because `public-url` is a misleading name.

This value configures the URL of the OIDC provider. How this URL will be used depends on the login flow that will be used (i.e. normal OIDC flow, or Authorization Code OIDC flow). In one case it needs to be public (users go there to login) and in another it can be private (used by AuthService to talk to OIDC provider (Dex) and can be an in-cluster K8s Service URL).

To summarize: We need to update the name/desc of this config and expose how it's used. Because it doesn't always have to be a public-url, and thus the current name is misleading.

Reply:
coming back to this since it recently became relevant again...

@jnsgruk's point is still correct about how this hard-codes the oidc provider (`dex-auth`) and namespace (`kubeflow`).  These are good guesses for our setup (the model **must** be `kubeflow` atm due to limitations upstream, but the oidc provider might be configurable) but not bulletproof.  Although since it is in a config, on the off chance someone needs to change it they can

Reply:
What we could do instead of defaulting to `http://dex-auth.kubeflow.svc.cluster.local:5556` is document the format this config option should have `http://<oidc-provider-domain-name>:<port(?)>`, change the name of the config option to `oidc-issuer-url` or something similar. 

In the charm code, we could have a new integration between dex-auth and oidc-gatekeepr where the latter constructs the `oidc-issuer-url` from the dex-auth Service in the absence of the config option.

Reply:
After speaking with @kimwnasptd about this config option, we may not even need it at all. In reality what happens is that there will always be a connection between oidc-gatekeeper and dex-auth, if there is any other identity provider, it has to connect through dex, so it always looks like oidc-gatekeeper <-> dex <-> id provider.

Perhaps the best approach here is to just let the oidc-gatekeeper to construct the issuer-url from the service name and port that dex-auth can share through relation data.